js/ui/widget/ui.widget.js

```diff
@@ -555,188 +516,169 @@ var Widget = DOMComponentWithTemplate.inherit({
                 this._attachFeedbackEvents();
                 break;
-            case "hoverStateEnabled":
+            case 'hoverStateEnabled':
                 this._attachHoverEvents();
                 break;
-            case "tabIndex":
-            case "focusStateEnabled":
+            case 'tabIndex':
+            case 'focusStateEnabled':
                 this._refreshFocusState();
                 break;
-            case "onFocusIn":
-            case "onFocusOut":
+            case 'onFocusIn':
+            case 'onFocusOut':
                 break;
-            case "accessKey":
+            case 'accessKey':
                 this._renderAccessKey();
                 break;
-            case "visible":
-                var visible = args.value;
-                this._toggleVisibility(visible);
+            case 'visible':
+                this._toggleVisibility(value);
                 if(this._isVisibilityChangeSupported()) {
                     // TODO hiding works wrong
-                    this._checkVisibilityChanged(args.value ? "shown" : "hiding");
+                    this._checkVisibilityChanged(value ? 'shown' : 'hiding');
                 }
                 break;
-            case "onKeyboardHandled":
+            case 'onKeyboardHandled':
                 this._attachKeyboardEvents();
                 break;
-            case "onContentReady":
+            case 'onContentReady':
                 this._initContentReadyAction();
                 break;
             default:
                 this.callBase(args);
         }
     },
 
-    _isVisible: function() {
-        return this.callBase() && this.option("visible");
+    _isVisible() {
+        const { visible } = this.option();
+
+        return this.callBase() && visible;
     },
 
-    beginUpdate: function() {
+    beginUpdate() {
         this._ready(false);
         this.callBase();
     },
 
-    endUpdate: function() {
+    endUpdate() {
         this.callBase();
 
         if(this._initialized) {
             this._ready(true);
         }
     },
 
-    _ready: function(value) {
+    _ready(value) {
         if(arguments.length === 0) {
             return this._isReady;
         }
 
         this._isReady = value;
-
     },
 
-    setAria: function() {
-        var setAttribute = function(option) {
-            var attrName = (option.name === "role" || option.name === "id") ? option.name : "aria-" + option.name,
-                attrValue = option.value;
-
-            if(typeUtils.isDefined(attrValue)) {
-                attrValue = attrValue.toString();
-            } else {
-                attrValue = null;
-            }
-
-            option.target.attr(attrName, attrValue);
-        };
-
-        if(!typeUtils.isPlainObject(arguments[0])) {
-            setAttribute({
-                name: arguments[0],
-                value: arguments[1],
-                target: arguments[2] || this._getAriaTarget()
-            });
+    setAria(...args) {
```

Comment:
```suggestion
    setAria(arg0, arg1, arg2) {
```
No need for spread

Reply:
Why? It looks shortly and hints that the number of parameters is not constant

Reply:
As I get from the code the number of args is constant, as they are accessed only by index, and not processed as collection.

Reply:
`setArea({}, target)` or `setArea(name, value, target)`

Reply:
I'll create new PR for this dispute